gnome-image-installer/gnome-image-installer.c

```diff
@@ -374,6 +375,11 @@ gis_page_get_type();
       gis_store_set_image_uuid (uuid);
     }
 
+  udisks_client = udisks_client_new_sync (NULL, &error);
+  if (udisks_client == NULL)
```

Comment:
In the previous incarnation, it's true that 2 out of 3 pages didn't check the return value, but `disktarget` did:

```
  priv->client = udisks_client_new_sync(NULL, &error);
  if (priv->client == NULL)
    {
      g_error("Unable to enumerate disks: %s", error->message);
      g_error_free(error);
    }
```

`g_error()` calls `abort()` so the `g_error_free` seems a bit optimistic ;) This branch does the same. It's true that doing this in `main()` rather than when the 2nd (3rd on `eosinstaller` images) page is shown makes it more likely to fail, given that you can't get past the `diskimage` page if `udisks_client_new_sync()` fails.

I don't think retrying once on each attempt to use the client will help: in the case where the language selection page is skipped, if the `diskimage` page can't use udisks, no images will be found, so it'll be impossible to go forwards; and you can't go backwards; so there's no way to cause a retry.

Two options, not mutually exclusive:

* A retry loop?
* An error dialog rather than an unceremonious `g_error`?